src/NSwag.CodeGeneration/Models/ParameterModelBase.cs

```diff
@@ -29,28 +29,66 @@ public abstract class ParameterModelBase
         /// <param name="allParameters">All parameters.</param>
         /// <param name="settings">The settings.</param>
         /// <param name="generator">The client generator base.</param>
+        /// <param name="valueGenerator">Settings used to generate default values.</param>
         protected ParameterModelBase(string parameterName, string variableName, string typeName,
-            SwaggerParameter parameter, IList<SwaggerParameter> allParameters, CodeGeneratorSettingsBase settings, IClientGenerator generator)
+            SwaggerParameter parameter, IList<SwaggerParameter> allParameters, CodeGeneratorSettingsBase settings,
+            IClientGenerator generator, ValueGeneratorBase valueGenerator )
         {
-            Type = typeName;
-            Name = parameterName;
-            VariableName = variableName;
-
             _allParameters = allParameters;
             _parameter = parameter;
             _settings = settings;
             _generator = generator;
+
+            Type = typeName;
+            ValueGenerator = valueGenerator;
+            Name = parameterName;
+            VariableName = variableName;
+
+            if ( !parameter.IsRequired && parameter.Default != null ) {
+                Default = DefaultToString( parameter );
+
+                // If a default value is supplied, then we can automatically set 
+                // it when using the controller interface indirection
+                TypeInControllerInterface = 
+                    Type.EndsWith( "?" ) ? Type.Substring( 0, Type.Length - 1 ) : Type;
+            }
+            else {
+                TypeInControllerInterface = Type;
+            }
+        }
+
+        /// <summary>
+        /// Value generator to use for mapping from <see cref="SwaggerParameter"/> to <see cref="string"/> value in the generated code.
+        /// </summary>
+        protected ValueGeneratorBase ValueGenerator { get; set; }
+
+        private string DefaultToString( SwaggerParameter parameter ) {
+            return ValueGenerator?.GetDefaultValue( parameter, false, parameter.ActualTypeSchema.Id, null, true );

```

Comment:
DefaultToString is now just a wrapper around the ValueGenerator.

Reply:
@RSuter i am not entirely sure i am feeding the generator the correct arguments here? It passed the test i had against the previous code base, but i would like to understand those parameters a bit better.